src/webview/wkwebview/mod.rs

```diff
@@ -603,6 +647,36 @@ impl InnerWebView {
           function_ptr as *mut _ as *mut c_void,
         );
 
+        if let Some(on_navigation_started_handler) = attributes.on_navigation_started_handler {
+          let on_navigation_started_handler = Box::into_raw(Box::new(Box::new(move || {
+            on_navigation_started_handler(url_from_webview(webview));
+          }) as Box<dyn Fn()>));
+          (*navigation_policy_handler).set_ivar(
+            "on_page_navigating_function",
+            on_navigation_started_handler as *mut _ as *mut c_void,
+          );
+        }
+
+        if let Some(on_page_loading_handler) = attributes.on_page_loading_handler {
+          let on_page_loading_handler = Box::into_raw(Box::new(Box::new(move || {
+            on_page_loading_handler(url_from_webview(webview));
+          }) as Box<dyn Fn()>));
+          (*navigation_policy_handler).set_ivar(
+            "on_page_loading_function",
+            on_page_loading_handler as *mut _ as *mut c_void,
+          );
+        }
+
+        if let Some(on_page_loaded_handler) = attributes.on_page_loaded_handler {
+          let on_page_loaded_handler = Box::into_raw(Box::new(Box::new(move || {
+            on_page_loaded_handler(url_from_webview(webview));
+          }) as Box<dyn Fn()>));
+          (*navigation_policy_handler).set_ivar(
+            "on_page_loaded_function",
+            on_page_loaded_handler as *mut _ as *mut c_void,
+          );
+        }
```

Comment:
this is leaking resources, you need to keep a pointer to these handlers and drop them in the Drop impl of `InnerWebView`

Reply:
This piece of code needs refactor, so I'd also suggest decoupling this logic from the `navigation_handler` logic, preferably in a module on its own, like `download.rs` module

Reply:
Okay, sorry this took so long. I've done the refactor and I believe resolved the leak.